File: csharp/leap/Leap.cs
using System;

public static class Leap
{
    public static bool IsLeapYear(int year)
    {
        // A leap year is every year whose number is evenly divisible by four,
        // except for years which are both divisible by 100 and not divisible
        // by 400
        return    (year % 4   == 0) &&
               !( (year % 100 == 0) && (year % 400 != 0) );
    }
}

File: go/leap/leap.go
// Check if a year is a leap year
package leap

// Check if a year is a leap year.
//
// A leap year is every year that is evenly divisible by 4,
//   except every year that is evenly divisible by 100,
//   unless the year is also evenly divisible by 400.
//
func IsLeapYear(year int) bool {
	return (year%4) == 0 && (year%100 != 0 || year%400 == 0)
}
